app/search/controller.ts

```diff
@@ -1,46 +1,91 @@
 import Store from '@ember-data/store';
+import { A } from '@ember/array';
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { waitFor } from '@ember/test-waiters';
-import { task } from 'ember-concurrency';
-import { taskFor } from 'ember-concurrency-ts';
 import { tracked } from '@glimmer/tracking';
+import { task, timeout } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
 import Media from 'ember-responsive';
 
 import MetadataPropertySearchModel from 'ember-osf-web/models/metadata-property-search';
+export interface Filter {
+    property: string;
+    value: string;
+}
 
 export default class SearchController extends Controller {
     @service store!: Store;
     @service media!: Media;
+    @service toast!: Toastr;
 
     queryParams = ['q', 'page', 'sort'];
-    @tracked q?: string;
-    @tracked page?: number;
-    @tracked sort?: string;
+    @tracked q?: string = '';
+    @tracked page?: number = 1;
+    @tracked sort?: string = '-relevance';
+    @tracked activeFilters = A<Filter>([]);
 
     @tracked propertySearch?: MetadataPropertySearchModel;
 
-    get showSidenavToggle() {
+    get showSidePanelToggle() {
         return this.media.isMobile || this.media.isTablet;
     }
 
+    get filterableProperties() {
+        if (!this.propertySearch) {
+            return [];
+        }
+        return this.propertySearch.get('searchResultPage');
+    }
+
     @action
     onKeyPress(event: KeyboardEvent) {
```

Comment:
IIRC, ember has `@enter` on an `<Input>` element that you may use so you don't have to catch all keyboard events here.